### ISSUE TITLE
Provider chaining support

### ISSUE THREAD

**Issue by r-inaldi** [sergiodxa/flagged#22]
### Additional Context

Any plans to support provider chaining? I have a use case which some points of the application have additional features or change exiting ones.

I've tried to implement a wrapper around `FlagsProvider` using `useFeatures` but it throws an error if there's no provider declared. I've wrapped with a try catch but heard that it's not the best practice for hooks.

Example:

```js
<FlagsProvider features={{ a: { enabled: true }, b: { enabled: true } }}>
  <FlagsProvider features={{ a: { enabled: false } }}>
    // .. here I'd receive a.enabled === false and b.enabled === true
  </FlagsProvider>  
</FlagsProvider>
```

I have three ideas:

1. Export the context; or
2. Remove throw error from `useFeatures`; or
3. Implement chaining by default inside the library.

Let me know if this could be something useful for you and I can open a PR.

**Comment by sergiodxa** [comment]
I don't yet understand why would you need this, also this could be solved user-land.

```js
import { FlagsProvider, useFeatures } from "flagged";

function ExtendFlagsProvider({ features, children }) {
  const currentFeatures = useFeatures();
  return (
    <FlagsProvider features={{ ...currentFeatures, ...features }}>
      {children}
    </FlagsProvider>
  )
}
```
You don't need access to the context itself, and with this ExtendFlagsProvider you could add more features to the current features in an specific part of the application, you will always need to have at least a single FlagsProvider with an empty object or array so you could use that as your `currentFeatures`.

**Comment by r-inaldi** [comment]
> I don't yet understand why would you need this, also this could be solved user-land.

Different features available for different parts of the application.

---

> You don't need access to the context itself, and with this ExtendFlagsProvider you could add more features to the current features in an specific part of the application, you will always need to have at least a single FlagsProvider with an empty object or array so you could use that as your currentFeatures.

I'd prefer to avoid a different provider for this because I want to able to remove the "main" one and rely on another without the need of changing providers.

---

Just asking for this because it's hard to handle errors caused by hooks. I think it's not really good to wrap with a try catch.

**Comment by sergiodxa** [comment]
I added this on https://github.com/sergiodxa/flagged/commit/233851ea7e15776b48684f66a057bf10aa98ceb1 and it's released as **v2.0.0**

**Comment by r-inaldi** [comment]
Nice! thanks :)
